### PR TITLE
BUG: one to any power is still 1.  Broken edgecase for int arrays

### PR DESCRIPTION
**Pull request by ewmoore**
Fixes gh-7648.

### CODE REVIEW

**Comment by mhvk** [comment]
Ah, looking back now I see this all comes from raising to some power without `pow`. But I think you are still missing one special case, namely for raising `-1` to some power.

**Comment by charris** [comment]
Thanks Eric.

**Comment by charris** [comment]
@mhvk Oops, indeed.

**Comment by charris** [comment]
@ewmoore I'd still like to see this whole thing discussed on the list. We have incompatible behavior between scalars and arrays, plus the change to raise an error for the ufunc for division by zero. Not that there is anything wrong, but because this is a messy corner with some nasty tradeoffs, I'd like to have some sort of consensus as to the proper behavior.

**Comment by mhvk** [comment]
Note that this still leaves `-1 ** (negative int)` incorrect!

**Comment by charris** [comment]
I posted to the list. Note that in 1.10 scalar arrays are not handled as arrays, but return int64. I think we need to propose the behavior we want and test *all* the integer types, together with +1, -1, 0. The tests do seem somewhat lacking at the moment.

File: numpy/core/src/umath/loops.c.src [hunk 1]
```diff
@@ -921,6 +921,10 @@ NPY_NO_EXPORT void
           *((@type@ *)op1) = 1;
           continue;
         }
+        if (in1 == 1) {
+            *((@type@ *)op1) = 1;
+            continue;
+        }
         if (in2 < 0 || in1 == 0) {
```
Comment (charris) [inline numpy/core/src/umath/loops.c.src]: How about drop the previous if and
```
if (in2 < 0 && (in1 > 1 || in1 < -1)) {
```
And a special check for divide by zero.

**Pull request merged**